format/bplist/bplist.go

```diff
@@ -0,0 +1,245 @@
+			d.FieldStrFn("value", func(d *decode.D) string {
+				v := d.F64()
+				t := time.Unix(int64(v), 0)
+				return t.String()
+			})
+		case elementTypeData:
+			d.FieldStruct("value", func(d *decode.D) {
+				n := decodeSize(d)
+				d.FieldRawLen("value", int64(n))
+			})
+		case elementTypeASCIIString:
+			d.FieldStruct("value", func(d *decode.D) {
+				d.FieldStrFn("value", func(d *decode.D) string {
+					n := decodeSize(d)
+					return d.UTF8(int(n))
+				})
+			})
+		case elementTypeUnicodeString:
+			d.FieldStrFn("value", func(d *decode.D) string {
+				n := decodeSize(d)
+				return d.UTF16(int(n))
+			})
+		case elementTypeUID:
+			n := decodeSize(d)
+			d.FieldUFn("value", func(d *decode.D) uint64 {
+				return d.UBigIntBE(int(n)).Uint64()
+			})
+		case elementTypeArray:
+			n := decodeSize(d)
+			i := uint64(0)
+			d.FieldStructArrayLoop("elements", "item",
+				func() bool { return i < n },
+				func(d *decode.D) {
+					idx := d.FieldU8("object_index")
+					decodeReference(d, p, idx)
+					i++
+				})
+		case elementTypeSet:
+			n := decodeSize(d)
+			i := uint64(0)
+			d.FieldArrayLoop("entries",
+				func() bool { return i < n },
+				func(d *decode.D) {
+					idx := d.FieldU8("object_index")
+					decodeReference(d, p, idx)
+					i++
+				})
+		case elementTypeDict:
+			d.FieldStruct("dictionary", func(d *decode.D) {
+				s := decodeSize(d)
+				i := uint64(0)
+				d.FieldStructArrayLoop("entries", "entry",
+					func() bool { return i < s },
+					func(d *decode.D) {
+						var ki, vi uint64
+						ki = d.FieldU8("key_index")
+						d.SeekRel(int64((s-1)*uint64(p.T.objRefSize)*8), func(d *decode.D) {
+							vi = d.FieldU8("value_index")
+						})
+						i++
+						d.FieldStruct("key", func(d *decode.D) {
+							decodeReference(d, p, ki)
+						})
+						d.FieldStruct("value", func(d *decode.D) {
+							decodeReference(d, p, vi)
+						})
+					})
+			})
+		default:
+			d.Errorf("unknown type marker: %d", m)
+		}
+	})
+
+	return nil
+}
+
+func decodeReference(d *decode.D, p *plist, idx uint64) {
+	d.SeekAbs(int64(p.O[idx]*8), func(d *decode.D) {
+		decodeItem(d, p)
+	})
+}
+
+type trailer struct {
+	offTblOffSize    int64
+	objRefSize       int64
+	nObjects         int64
+	topObjectOffset  int64
+	offsetTableStart int64
+}
+
+type plist struct {
+	T *trailer
+	O []uint64
+}
+
+func bplistDecode(d *decode.D, _ any) any {
+	d.FieldStruct("header", func(d *decode.D) {
+		d.FieldUTF8("magic", 6, d.AssertStr("bplist"))
+		d.FieldUTF8("version", 2, d.AssertStr("00"))
+	})
+
+	t := new(trailer)
+
+	d.SeekAbs(d.Len()-32*8, func(d *decode.D) {
+		d.FieldStruct("trailer", func(d *decode.D) {
+			d.FieldU40("padding") // unused
+			d.FieldS8("sort_version")
+			t.offTblOffSize = d.FieldS8("offset_table_offset_size")
+			t.objRefSize = d.FieldS8("object_reference_size")
+			t.nObjects = d.FieldS64BE("object_count")
+			t.topObjectOffset = d.FieldS64BE("top_object_offset")
+			t.offsetTableStart = d.FieldS64BE("offset_table_start")
```

Comment:
With different default endian BE could be skipped, assuming there is no or few LE encoded values